### PR TITLE
PMM-6653 Copy pxc/v1 and psmdb/v1 objects definitions

### PR DESCRIPTION
**Pull request by percona-csalguero**

### CODE REVIEW

File: k8_api/pxc/v1/pxc_prestore_types.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	// RestoreStartCluster starting cluster.
+	RestoreStartCluster BcpRestoreStates = "Starting Cluster"
+	// RestoreFailed restore has failed.
+	RestoreFailed BcpRestoreStates = "Failed"
+	// RestoreSucceeded restore completed OK.
+	RestoreSucceeded BcpRestoreStates = "Succeeded"
+)
+
+// CheckNsetDefaults checks for valid defaults.
+func (cr *PerconaXtraDBClusterRestore) CheckNsetDefaults() error {
```
Comment (BupycHuk) [inline k8_api/pxc/v1/pxc_prestore_types.go:89]: Do we use this method?

File: service/k8sclient/k8client.go [hunk 2]
```diff
@@ -22,20 +22,19 @@
 	"encoding/json"
 
 	"github.com/AlekSi/pointer"
+	pxc "github.com/percona-platform/dbaas-controller/k8_api/pxc/v1"
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply service/k8sclient/k8client.go:25]: it's a right message. Please move all local imports to the third block.

**Review by BupycHuk — CHANGES_REQUESTED** [review]
There is a lot of methods which is not used, do we really need them?

**Review by askomorokhov — APPROVED** [review]

**Comment by percona-csalguero** [comment]
@BupycHuk I removed all method reported as unused by enabling unused linter. Could you re review?

File: k8_api/pxc/v1/pxc_types.go [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package v1
+
+import (
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	"github.com/percona/percona-xtradb-cluster-operator/version"
```
Comment (BupycHuk) [inline k8_api/pxc/v1/pxc_types.go:21]: we still have dependency on this

**Review by BupycHuk — CHANGES_REQUESTED** [review]

**Comment by BupycHuk** [comment]
Tests are failing

File: go.mod [hunk 4]
```diff
@@ -14,35 +14,55 @@ go 1.14
...
+	github.com/jetstack/cert-manager v1.0.0-beta.0
+	github.com/klauspost/compress v1.11.1 // indirect
+	github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369 // indirect
+	github.com/percona-platform/dbaas-api v0.0.0-20201026075036-655507121a69
 	github.com/percona-platform/saas v0.0.0-20201008124851-3c2c6c2ec0ce
-	github.com/percona/percona-backup-mongodb v1.2.0 // indirect
-	github.com/percona/percona-server-mongodb-operator v1.4.0
-	github.com/percona/percona-xtradb-cluster-operator v1.4.0
+	github.com/percona/percona-backup-mongodb v1.3.2 // indirect
+	github.com/percona/percona-server-mongodb-operator v1.5.0
```
Comment (AlekSi) [inline go.mod:35]: > Expected result: dbaas-controller does not depend on github.com/percona/percona-xtradb-cluster-operator and github.com/percona/percona-server-mongodb-operator Go modules.

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: go.mod (hunk 4 above)
Comment (BupycHuk) [reply go.mod:35]: percona-server-mongodb-operator is removed in the other PR

**Review by BupycHuk — APPROVED** [review]

File: go.mod (hunk 4 above)
Comment (AlekSi) [reply go.mod:35]: So I guess we should merge the other PR first? Because this ticket is not done otherwise

Comment (AlekSi) [reply go.mod:35]: The "other PR" in question: https://github.com/percona-platform/dbaas-controller/pull/67

Comment (BupycHuk) [reply go.mod:35]: Actually @percona-csalguero can remove one line from imports and this PR should be ready. No need to wait for the second PR.

Comment (AlekSi) [reply go.mod]: Did that in 67ba54db1d3a665e646a80204e2cf7e6b4ecfcbd, but that's not enough: we still depend on k8s packages, and on different versions than operators. We should copy a trimmed version of API machinery too.

Comment (AlekSi) [reply go.mod]: https://jira.percona.com/browse/PMM-6830

**Comment by AlekSi** [comment]
And there are merge conflicts now

File: k8_api/pxc/v1/pxc_prestore_types.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+type PerconaXtraDBClusterRestoreStatus struct {
+	State         BcpRestoreStates `json:"state,omitempty"`
+	Comments      string           `json:"comments,omitempty"`
+	CompletedAt   *metav1.Time     `json:"completed,omitempty"`
+	LastScheduled *metav1.Time     `json:"lastscheduled,omitempty"`
+}
+
+// PerconaXtraDBClusterRestore is the Schema for the perconaxtradbclusterrestores API.
+type PerconaXtraDBClusterRestore struct {
+	metav1.TypeMeta   `json:",inline"`
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply k8_api/pxc/v1/pxc_prestore_types.go:40]: Checked it and it works. It encodes all fields of `metav1.TypeMeta` as fields of `PerconaXtraDBClusterRestore` in json.

**Review by BupycHuk — APPROVED** [review]

File: k8_api/pxc/v1/pxc_prestore_types.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (AlekSi) [reply k8_api/pxc/v1/pxc_prestore_types.go:40]: How? Where is the code that handles that?

**Pull request merged**